### ISSUE TITLE
Support reading init_script file which is not under resource/ directory

### ISSUE THREAD

**Issue by flycash** [apache/incubator-shenyu#1604]
## Feature Request

For now, when shenyu-admin start up, it will run the `init_script`. But it use `org.apache.ibatis.io.Resources#getResourceAsReader(java.lang.String)` to read the file which means that it only read those files under `resource` directory

`org.apache.shenyu.admin.spring.LocalDataSourceLoader`:

![image](https://user-images.githubusercontent.com/9923838/121541426-df1b7f80-ca39-11eb-9397-46c4a3b3e85b.png)

#### Is your feature request related to a problem? Please describe

I use the docker to start admin-dist and here is part of my docker-compose file:
```yml
  shenyu-admin:
    image: dromara/shenyu-admin:2.3.1-SNAPSHOT
    container_name: shenyu-admin
    restart: always
    networks:
      - shenyu
    volumes:
      - type: bind
        source: ./admin-config
        target: /opt/admin-config/
    depends_on:
#      - third-party-ready
      shenyu-mysql:
        condition: service_healthy
#      - redis
#      - zookeeper
#      - prometheus
    ports:
      - 9095:9095
    environment:
      - SPRING_PROFILES_ACTIVE=mysql
      - "spring.datasource.url=jdbc:mysql://shenyu-mysql:3306/shenyu?useUnicode=true&characterEncoding=utf-8&useSSL=false"
      - "spring.datasource.password=root"
      - "shenyu.database.init_script=/opt/admin-config/schema.sql"
    healthcheck:
      test: ["CMD", "wget", "http://shenyu-admin:9095"]
      timeout: 20s
      retries: 10
```
When I start the docker compose, I got error:
![image](https://user-images.githubusercontent.com/9923838/121540605-33722f80-ca39-11eb-9f49-042b5a18675f.png)

And I can't find any solution that could bind the file to `schema.sql` which is packaged inside the `shenyu-admin.jar`, or add new files into existing `shenyu-admin.jar`

![image](https://user-images.githubusercontent.com/9923838/121541983-50f3c900-ca3a-11eb-9590-18cf945b6aaa.png)

#### Describe the solution you'd like

My proposal is that support reading `init_script` from anywhere. 

But the key point is to keep compatibility. 

#### Describe alternatives you've considered

Users who has similar requirement could build the `admin-dist` by themselves. 

But it means that users must clone the source code, install maven, install docker, and then learn how to build it. 

If want to test their `init_script.sql`, they have to build it again and again.
 
#### Additional context

Please read the [Contribution Guideline](https://dromara.org/projects/soul/contributor/) before submitting the PR

**Comment by yu199195** [comment]
Hi,  We don't have a custom path to read the init script, can someone from the community fix it?

**Comment by zhaiweij** [comment]
I understand your needs and I will improve

**Comment by hermeshephaestus** [comment]
@yu199195 Is anyone currently working on it; if not, assign the task to me!

**Comment by flycash** [comment]
Now it has been resolved,, check #1690
